nixos/modules/system/etc/make-etc.sh

```diff
@@ -2,43 +2,40 @@ source $stdenv/setup
 
 mkdir -p $out/etc
 
-set -f
-sources_=($sources)
-targets_=($targets)
-modes_=($modes)
-users_=($users)
-groups_=($groups)
-set +f
-
-for ((i = 0; i < ${#targets_[@]}; i++)); do
-    source="${sources_[$i]}"
-    target="${targets_[$i]}"
+readarray -t items < <(jq -c '.[]' < "$data")
+
+for item in "${items[@]}"; do
+    source="$(jq -nr '$item.source' --argjson item "$item")"
+    target="$(jq -nr '$item.target' --argjson item "$item")"
+    mode="$(jq -nr '$item.mode' --argjson item "$item")"
+    user="$(jq -nr '$item.user' --argjson item "$item")"
+    group="$(jq -nr '$item.group' --argjson item "$item")"
 
     if [[ "$source" =~ '*' ]]; then
 
         # If the source name contains '*', perform globbing.
```

Comment:
The special handling for glob patterns here should probably be either documented or dropped.
People might be surprised if they files in /etc with a * in them..